app/src/main/java/com/lucas/medicaltools/MedicalToolAdapter.kt

```diff
@@ -0,0 +1,93 @@
+package com.lucas.medicaltools
+
+import android.content.Context
+import android.graphics.Color
+import android.net.Uri
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.*
+import androidx.core.content.contentValuesOf
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.snackbar.Snackbar
+import com.lucas.medical_equip.repository.MedicalTool
+import com.lucas.medicaltools.databinding.MedicalToolItemBinding
+import com.squareup.picasso.Picasso
+import java.util.*
+import kotlin.collections.ArrayList
+
+class MedicalToolAdapter(private var medicalTools: List<MedicalTool>) : RecyclerView.Adapter<MedicalToolAdapter.ViewHolder>(), Filterable {
+    var medicalToolsFiltered = ArrayList<MedicalTool>()
+
+    init {
+        medicalToolsFiltered = medicalTools as ArrayList<MedicalTool>
+    }
+    inner class ViewHolder(val binding: MedicalToolItemBinding) : RecyclerView.ViewHolder(binding.root)
+
+        // val medicalToolImage = itemView.findViewById<ImageView>(R.id.medical_tool_image)
+        // val medicalToolDescription = itemView.findViewById<TextView>(R.id.medical_tool_text)
+        // val isAvailable = itemView.findViewById<TextView>(R.id.medical_tool_avail)
+
+
+
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+
+        val context = parent.context
+        val inflater = LayoutInflater.from(context)
+        val binding = MedicalToolItemBinding.inflate(inflater, parent, false)
+
+        return ViewHolder(binding)
+    }
+
+    override fun getItemCount(): Int {
+        return medicalToolsFiltered.size
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        //Get data model based on position
+        val medicalTool: MedicalTool = medicalToolsFiltered.get(position)
```

Comment:
You'll not have a filtered list once we removed the filterable, adapter will have a single data list that it renders and isn't involved in any filtering